### ISSUE TITLE
Parameters containing percent-encoded special characters are not decoded properly

### ISSUE THREAD

**Issue by dittos** [rackt/react-router#650]
When I run the code:

```js
var App = React.createClass({
    mixins: [ReactRouter.State],
    render: function() {
        return <span>{this.getParams().title}</span>;
    }
});

ReactRouter.run(<ReactRouter.Route path="/:title" handler={App} />, function(Handler) {
    React.render(<Handler />, document.body);
});

location.href = '#/asdf%26qwer';
```

Expected output is `asdf&qwer`, but `asdf%26qwer` is actually rendered. Test with [jsfiddle](http://jsfiddle.net/qn68sdg6/3/). (Reproducible with Chrome/Firefox/IE)

Maybe `decodeURIComponent` instead of `decodeURI` should be used in `Path.decode` function.

**Comment by mjackson** [comment]
@dittos This should be fixed in master. Can you please verify?

**Comment by mjackson** [comment]
This should be fixed in db2607d4d33322acb2c77c8906068015f2254c12. Please feel free to re-open if not.
